ci/test_cudf_polars_polars_tests.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -eou pipefail
+
+# We will only fail these tests if the PR touches code in pylibcudf
+# or cudf_polars itself.
+# Note, the three dots mean we are doing diff between the merge-base
+# of upstream and HEAD. So this is asking, "does _this branch_ touch
+# files in cudf_polars/pylibcudf", rather than "are there changes
+# between upstream and this branch which touch cudf_polars/pylibcudf"
+# TODO: is the target branch exposed anywhere in an environment variable?
```

Comment:
Yes, in GHA it is `GITHUB_BASE_REF`: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

Reply:
Thanks, done.

Reply:
Seemed like that didn't work: https://github.com/rapidsai/cudf/actions/runs/10705555021/job/29682172407?pr=16710#step:8:19

I will go back to hard-coding for now.